### ISSUE TITLE
Use php in smarty tpl file

### ISSUE THREAD

**Issue by trodat330** [formtools/core#518]
I'm trying to run some php inside the smarty tpl file that calculates a persons age based on their provided date of birth. In Form Tools administration, under "Modules > Export Manager > HTML / Printer-friendly > Export Types" is where I need to add the age calculation php code.

Age: {php}
/**
 * Simple PHP age Calculator
 * 
 * Calculate and returns age based on the date provided by the user.
 * @param   date of birth('Format:yyyy-mm-dd').
 * @return  age based on date of birth
 */
function ageCalculator($dob){
    if(!empty($dob)){
        $birthdate = new DateTime($dob);
        $today   = new DateTime('today');
        $age = $birthdate->diff($today)->y;
        return $age;
    }else{
        return 0;
    }
}
$dob = 'SOMETHING';
echo ageCalculator($dob);
{/php}

Basically, it's a baseball player submission and on the form, it's called <?=@$fields['Player1_DOB']?>.

In the php above, the $dob = 'SOMETHING'; is where the date of birth should be displayed. But, I cannot get it to work. I tried {Player1_DOB} and every variation of it, but nothing seems to be working. When I put an actual date like "2010/01/01" it works fine.

**Comment by benkeen** [comment]
Hey @trodat330, sorry for not responding sooner. 

So just so I'm clear, you mean the $dob = 'SOMETHING'; line is where you want to extract the value of the something from the Smarty template? 

Which Smarty template? 

Couple of things to help locate what you're looking for. 
- the `{debug}` function is handy to drop into any Smarty template. That'll open up a window listing all the variables available in the scope of whereever you dropped it. 
- inside a `{php}` block, try adding: `print_r(get_defined_vars());` - that does something rather like the previous line. It'll output all variables defined within the scope. 

I did a quick test within an Export Manager -> Export Type template to see how to extract a particular form field value and frankly it ain't pretty! But I found that this line allowed me to get access to a specific field value in my form:

```
echo $_smarty_tpl->smarty->tpl_vars["submissions"]->value[0]["col_1"];
```

(where col_1 was the column name of my field).

**Comment by trodat330** [comment]
Hi Ben,
That is correct. I need to replace 'SOMETHING' with the DOB value. So, I  tried this:

{php}
/**

Simple PHP age Calculator
Calculate and returns age based on the date provided by the user.
@param date of birth('Format:yyyy-mm-dd').
@return age based on date of birth
*/
function ageCalculator($dob){
if(!empty($dob)){
$birthdate = new DateTime($dob);
$today = new DateTime('today');
$age = $birthdate->diff($today)->y;
return $age;
}else{
return 0;
}
}
$dob = 'echo $_smarty_tpl->smarty->tpl_vars["submissions"]->value[0]["player1_dob"];';
echo ageCalculator($dob);
{/php}

But, that didn't work.
